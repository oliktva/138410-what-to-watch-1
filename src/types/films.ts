import PropTypes from 'prop-types';

export const filmPropTypes = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  video: PropTypes.string.isRequired
});

export const filmsPropTypes = PropTypes.arrayOf(filmPropTypes.isRequired);

export interface FilmProps {
  name: string;
  img: string;
  video: string;
  id: number;
}
