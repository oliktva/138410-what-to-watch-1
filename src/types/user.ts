import PropTypes from 'prop-types';

export const UserPropTypes = PropTypes.shape({
  id: PropTypes.number,
  email: PropTypes.string,
  name: PropTypes.string,
  avatarUrl: PropTypes.string,
});

export interface UserProps {
  id: number | null;
  email: string | null;
  name: string | null;
  avatarUrl: string | null;
}
