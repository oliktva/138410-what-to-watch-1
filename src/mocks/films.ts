import {FilmProps} from "src/types/films";

export const films: FilmProps[] = [{
  id: 1,
  name: `Fantastic Beasts`,
  img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 2,
  name: `Bohemian Rhapsody`,
  img: `img/bohemian-rhapsody.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 3,
  name: `Macbeth`,
  img: `img/macbeth.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 4,
  name: `Aviator`,
  img: `img/aviator.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 5,
  name: `We need to talk about Kevin`,
  img: `img/we-need-to-talk-about-kevin.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 6,
  name: `What We Do in the Shadows`,
  img: `img/what-we-do-in-the-shadows.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 7,
  name: `Revenant`,
  img: `img/revenant.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 8,
  name: `Johnny English`,
  img: `img/johnny-english.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 9,
  name: `Shutter Island`,
  img: `img/shutter-island.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 10,
  name: `Pulp Fiction`,
  img: `img/pulp-fiction.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 11,
  name: `No Country for Old Men`,
  img: `img/no-country-for-old-men.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 12,
  name: `Snatch`,
  img: `img/snatch.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 13,
  name: `Moonrise Kingdom`,
  img: `img/moonrise-kingdom.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 14,
  name: `Seven Years in Tibet`,
  img: `img/seven-years-in-tibet.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 15,
  name: `Midnight Special`,
  img: `img/midnight-special.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 16,
  name: `War of the Worlds`,
  img: `img/war-of-the-worlds.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 17,
  name: `Dardjeeling Limited`,
  img: `img/dardjeeling-limited.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 18,
  name: `Orlando`,
  img: `img/orlando.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 19,
  name: `Mindhunter`,
  img: `img/mindhunter.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 20,
  name: `Midnight Special`,
  img: `img/midnight-special.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}];
