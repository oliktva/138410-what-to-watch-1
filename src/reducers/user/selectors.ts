import namespaces from 'src/reducers/namespaces';

import {State} from 'src/types/reducer';
import {UserProps} from 'src/types/user';

export const getUser = (state: State): UserProps => {
  const {id, email, name, avatarUrl} = state[namespaces.USER];

  return {id, email, name, avatarUrl};
};

export const getError = (state: State): (string | undefined) => {
  return state[namespaces.USER].error;
};

export const getIsAuthorizationRequired = (state: State): boolean => {
  return state[namespaces.USER].isAuthorizationRequired;
};
