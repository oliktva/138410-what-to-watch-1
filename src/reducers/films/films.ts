import {Action as ReduxAction} from 'redux';
import {ThunkDispatch as ReduxThunkDispatch, ThunkAction as ReduxThunkAction} from 'redux-thunk';
import {AxiosResponse, AxiosInstance} from 'axios';
import {toCamel} from 'convert-keys';

import {GenreProps} from 'src/types/genres';
import {FilmProps} from 'src/types/films';
import {ReviewProps} from 'src/types/reviews';
import {ThunkDispatch, ThunkAction, State as AppState} from 'src/types/reducer';

export const SET_GENRE = `SET_GENRE`;
export const LOAD_FILMS_SUCCESS = `LOAD_FILMS_SUCCESS`;
export const LOAD_FAVORITE_SUCCESS = `LOAD_FAVORITE_SUCCESS`;
export const UPDATE_FILM = `UPDATE_FILM`;
export const LOAD_PROMO = `LOAD_PROMO`;
export const LOAD_REVIEWS = `LOAD_REVIEWS`;

const ALL_GENRES = `All genres`;

export interface State {
  genre: GenreProps;
  items: FilmProps[];
  favorite: FilmProps[];
  promo: FilmProps | null;
  reviews: {[key: number]: ReviewProps[]} | null;
}

interface SetFilterAction extends ReduxAction {
  type: typeof SET_GENRE;
  payload: GenreProps;
}

interface LoadFilmsAction extends ReduxAction {
  type: typeof LOAD_FILMS_SUCCESS;
  payload: FilmProps[];
}

interface LoadFavoritesAction extends ReduxAction {
  type: typeof LOAD_FAVORITE_SUCCESS;
  payload: FilmProps[];
}

interface UpdateFilmAction extends ReduxAction {
  type: typeof UPDATE_FILM;
  payload: FilmProps;
}

interface LoadFilmPromoAction extends ReduxAction {
  type: typeof LOAD_PROMO;
  payload: FilmProps;
}

interface LoadReviewsAction extends ReduxAction {
  type: typeof LOAD_REVIEWS;
  payload: {id: number; reviews: ReviewProps[]};
}

export type Action =
  SetFilterAction | LoadFilmsAction | LoadFavoritesAction | UpdateFilmAction | LoadFilmPromoAction | LoadReviewsAction;

export type ThunkDispatch = ReduxThunkDispatch<State, AxiosInstance, Action>;
export type ThunkAction = ReduxThunkAction<Promise<void>, State, AxiosInstance, Action>;

export const initialState: State = {
  genre: ALL_GENRES,
  items: [],
  favorite: [],
  promo: null,
  reviews: null
};

const updateFilm = (film: FilmProps, filmsList: FilmProps[]): FilmProps[] =>
  filmsList.map((f: FilmProps): FilmProps => (f.id === film.id ? film : f));

export const ActionCreator = {
  setFilterByGenre: (genre: GenreProps): SetFilterAction => {
    return {
      type: SET_GENRE,
      payload: genre
    };
  },

  loadFilms: (films: FilmProps[]): LoadFilmsAction => {
    return {
      type: LOAD_FILMS_SUCCESS,
      payload: films
    };
  },

  loadFavorite: (films: FilmProps[]): LoadFavoritesAction => {
    return {
      type: LOAD_FAVORITE_SUCCESS,
      payload: films
    };
  },

  updateFilm: (film: FilmProps): UpdateFilmAction => {
    return {
      type: UPDATE_FILM,
      payload: film
    };
  },

  loadPromo: (film: FilmProps): LoadFilmPromoAction => {
    return {
      type: LOAD_PROMO,
      payload: film
    };
  },

  loadReviews: (id: number, reviews: ReviewProps[]): LoadReviewsAction => {
    return {
      type: LOAD_REVIEWS,
      payload: {id, reviews}
    };
  }
};

export const Operation = {
  loadFilms: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/films`)
        .then((response: AxiosResponse<Record<string, any>[]>): void => {
          const data = response.data.map((f): FilmProps => toCamel<FilmProps>(f));

          dispatch(ActionCreator.loadFilms(data));
        });
    };
  },

  loadFavorite: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/favorite`)
        .then((response: AxiosResponse<Record<string, any>[]>): void => {
          const data = response.data.map((f): FilmProps => toCamel<FilmProps>(f));

          dispatch(ActionCreator.loadFavorite(data));
        });
    };
  },

  addToFavorites: (filmId: number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/favorite/${filmId}/1`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<FilmProps>(response.data);

          dispatch(ActionCreator.updateFilm(data));
        });
    };
  },

  removeFromFavorites: (filmId: number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/favorite/${filmId}/0`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<FilmProps>(response.data);

          dispatch(ActionCreator.updateFilm(data));
        });
    };
  },

  loadPromo: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/films/promo`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<FilmProps>(response.data);

          dispatch(ActionCreator.loadPromo(data));
        });
    };
  },

  loadReviews: (id: number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/comments/${id}`)
        .then((response: AxiosResponse<Record<string, any>[]>): void => {
          const data = response.data.map((r): ReviewProps => toCamel<ReviewProps>(r));

          dispatch(ActionCreator.loadReviews(id, data));
        });
    };
  },

  addReview: (id: number, reviewText: string, rating: string | number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/comments/${id}`, {
        comment: reviewText,
        rating: typeof rating === `string` ? parseInt(rating, 10) : rating
      }).then((response: AxiosResponse<Record<string, any>[]>): void => {
        const data = response.data.map((r): ReviewProps => toCamel<ReviewProps>(r));

        dispatch(ActionCreator.loadReviews(id, data));
      });
    };
  }
};

const reducer = (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case LOAD_FILMS_SUCCESS:
      return {
        ...state,
        items: action.payload
      };

    case SET_GENRE:
      return {
        ...state,
        genre: action.payload
      };

    case LOAD_FAVORITE_SUCCESS:
      return {
        ...state,
        favorite: action.payload
      };

    case UPDATE_FILM:
      return {
        ...state,
        promo: state.promo ? updateFilm(action.payload, [state.promo])[0] : state.promo,
        items: updateFilm(action.payload, state.items),
        favorite: updateFilm(action.payload, state.favorite)
      };

    case LOAD_PROMO: {
      return {
        ...state,
        promo: action.payload
      };
    }

    case LOAD_REVIEWS: {
      return {
        ...state,
        reviews: {
          ...state.reviews,
          [action.payload.id]: action.payload.reviews
        }
      };
    }

    default:
      return state;
  }
};

export default reducer;
