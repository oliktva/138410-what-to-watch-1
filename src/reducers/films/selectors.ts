import {createSelector} from 'reselect';

import Namespaces from 'src/reducers/namespaces';

import {State} from 'src/types/reducer';
import {FilmProps} from 'src/types/films';
import {GenreProps} from 'src/types/genres';

const ALL_GENRES = `All genres`;

export const getFilms = (state: State): FilmProps[] => {
  return state[Namespaces.FILMS].items;
};

export const getGenre = (state: State): GenreProps => {
  return state[Namespaces.FILMS].genre;
};

export const getFilmsByGenre = createSelector(
  getFilms,
  getGenre,
  (films: FilmProps[], genre: GenreProps): FilmProps[] => {
    return genre === ALL_GENRES ?
      films :
      films.filter((f: FilmProps): boolean => f.genre === genre);
  }
);

export const getGenres = createSelector(
  getFilms,
  (films: FilmProps[]): GenreProps[] => {
    const filmsGenres = films.map((f: FilmProps): GenreProps => f.genre)
      .reduce((result: GenreProps[], genre: GenreProps): GenreProps[] => {
        if (!result.includes(genre)) {
          result.push(genre);
        }

        return result;
      }, []);

    return [ALL_GENRES, ...filmsGenres];
  }
);
