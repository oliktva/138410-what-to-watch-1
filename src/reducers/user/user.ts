import {Action as ReduxAction} from 'redux';
import {AxiosResponse, AxiosError, AxiosInstance} from 'axios';
import {toCamel} from 'convert-keys';

import {UserProps} from 'src/types/user';
import {ThunkDispatch, ThunkAction, State as AppState} from 'src/types/reducer';

export const TOGGLE_AUTHORIZATION_REQUIRED = `TOGGLE_AUTHORIZATION_REQUIRED`;
export const LOG_IN_USER = `LOG_IN_USER`;
export const LOG_IN_USER_ERROR = `LOG_IN_USER_ERROR`;

export interface State {
  isAuthorizationRequired: boolean;
  id?: number;
  email?: string;
  name?: string;
  avatarUrl?: string;
  error?: string;
}

interface LogInUserAction extends ReduxAction {
  type: typeof LOG_IN_USER;
  payload: UserProps;
}

interface LogInUserErrorAction extends ReduxAction {
  type: typeof LOG_IN_USER_ERROR;
  payload: string;
}

export type Action = LogInUserAction | LogInUserErrorAction;

export const initialState: State = {
  isAuthorizationRequired: false,
  id: undefined,
  email: undefined,
  name: undefined,
  avatarUrl: undefined,
  error: undefined
};

export const ActionCreator = {
  logInUser: (user: UserProps): LogInUserAction => {
    return {
      type: LOG_IN_USER,
      payload: user
    };
  },

  logInUserError: (error: string): LogInUserErrorAction => {
    return {
      type: LOG_IN_USER_ERROR,
      payload: error
    };
  }
};

export const Operation = {
  loadUser: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/login`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<UserProps>(response.data);

          dispatch(ActionCreator.logInUser(data));
        }).catch((): void => {
          // don't care
        });
    };
  },

  logInUser: (email: string, password: string): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/login`, {
        email,
        password
      }).then((response: AxiosResponse<Record<string, any>>): void => {
        const data = toCamel<UserProps>(response.data);

        dispatch(ActionCreator.logInUser(data));
      }).catch((error: AxiosError): void => {
        dispatch(ActionCreator.logInUserError(error.message));
      });
    };
  }
};

const reducer = (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case LOG_IN_USER:
      return {
        ...state,
        ...action.payload,
        error: undefined
      };

    case LOG_IN_USER_ERROR:
      return {
        ...state,
        error: action.payload
      };

    default:
      return state;
  }
};

export default reducer;
