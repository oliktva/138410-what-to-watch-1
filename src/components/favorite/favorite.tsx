import React, {ComponentClass, ReactElement, PureComponent} from 'react';
import {connect, MapStateToProps, MapDispatchToProps} from 'react-redux';
import {compose} from 'recompose';

import needLogin from 'src/hocs/need-login/need-login';
import {getFavorite} from 'src/reducers/films/selectors';

import PageWrapper from 'src/components/page-wrapper/page-wrapper';
import Header from 'src/components/header/header';
import Footer from 'src/components/footer/footer';
import SmallMovieCard from 'src/components/small-movie-card/small-movie-card';

import {UserProps, UserPropTypes} from 'src/types/user';
import {State, ThunkDispatch} from 'src/types/reducer';
import {Operation} from 'src/reducers/films/films';
import {FilmProps} from 'src/types/films';

interface StateProps {
  favorite: FilmProps[];
}

interface DispatchProps {
  loadFavorite: () => Promise<void>;
}

interface OwnProps {
  user: UserProps;
}

type Props = StateProps & DispatchProps & OwnProps;

class Favorite extends PureComponent<Props> {
  public static propTypes = {
    user: UserPropTypes.isRequired
  }

  private _renderFilm(film: FilmProps): ReactElement {
    return (
      <SmallMovieCard
        key={film.id}
        film={film}
        isPlaying={false}
      />
    );
  }

  public render(): ReactElement {
    const {user, favorite} = this.props;

    return (
      <PageWrapper>
        <div className="user-page">
          <Header user={user} className="user-page__head" heading="My list" />
          <section className="catalog">
            <h2 className="catalog__title visually-hidden">Catalog</h2>
            <div className="catalog__movies-list">
              {favorite.map((film: FilmProps): ReactElement => {
                return this._renderFilm(film);
              })}
            </div>
          </section>
          <Footer />
        </div>
      </PageWrapper>
    );
  }
}

export {Favorite};

const mapStateToProps: MapStateToProps<StateProps, OwnProps, State> = (state: State): StateProps => ({
  favorite: getFavorite(state)
});

const mapDispatchToProps: MapDispatchToProps<DispatchProps, OwnProps> = (dispatch: ThunkDispatch): DispatchProps => ({
  loadFavorite: (): Promise<void> => dispatch(Operation.loadFavorite())
});

const ConnectedComponent: any =
  compose<Props, ComponentClass<OwnProps>>(
    connect<StateProps, DispatchProps, OwnProps, State>(mapStateToProps, mapDispatchToProps),
    needLogin
  )(Favorite);

export default ConnectedComponent;
