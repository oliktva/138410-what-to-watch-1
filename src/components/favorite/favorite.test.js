import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import {Favorite} from './favorite';

import {films} from 'src/fixtures/films';
import {user} from 'src/fixtures/user';

Enzyme.configure({adapter: new Adapter()});

it(`renders correctly`, () => {
  const favorites = shallow(
    <Favorite
      user={user}
      favorite={films}
      loadFavorite={jest.fn()}
    />
  );

  expect(toJson(favorites)).toMatchSnapshot();
});
