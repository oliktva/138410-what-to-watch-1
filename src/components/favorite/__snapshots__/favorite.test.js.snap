// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<PageWrapper>
  <div
    className="user-page"
  >
    <Header
      className="user-page__head"
      heading="My list"
      isLightLogo={true}
      user={
        Object {
          "avatarUrl": "img/1.png",
          "email": "Oliver.conner@gmail.com",
          "id": 1,
          "name": "Oliver.conner",
        }
      }
      withBreadcrumbs={false}
    />
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <div
        className="catalog__movies-list"
      >
        <SmallMovieCard
          film={
            Object {
              "backgroundColor": "#795433",
              "backgroundImage": "https://es31-server.appspot.com/wtw/static/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "genre": "Crime",
              "id": 1,
              "isFavorite": false,
              "name": "Pulp Fiction",
              "posterImage": "https://es31-server.appspot.com/wtw/static/film/poster/Pulp_Fiction.jpg",
              "previewImage": "https://es31-server.appspot.com/wtw/static/film/preview/pulp-fiction.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": 8.9,
              "released": 1994,
              "runTime": 153,
              "scoresCount": 1635992,
              "starring": Array [
                "John Travolta",
                "Uma Thurman",
                "Samuel L. Jackson",
              ],
              "videoLink": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            }
          }
          isPlaying={false}
          key="1"
        />
        <SmallMovieCard
          film={
            Object {
              "backgroundColor": "#9B7E61",
              "backgroundImage": "https://es31-server.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg",
              "description": "As Earth is invaded by alien tripod fighting machines, one family fights for survival.",
              "director": "Steven Spielberg",
              "genre": "Adventure",
              "id": 2,
              "isFavorite": false,
              "name": "War of the Worlds",
              "posterImage": "https://es31-server.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg",
              "previewImage": "https://es31-server.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": 6.5,
              "released": 2005,
              "runTime": 116,
              "scoresCount": 386834,
              "starring": Array [
                "Tom Cruise",
                "Dakota Fanning",
                "Tim Robbins",
              ],
              "videoLink": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
            }
          }
          isPlaying={false}
          key="2"
        />
        <SmallMovieCard
          film={
            Object {
              "backgroundColor": "#A6B7AC",
              "backgroundImage": "https://es31-server.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg",
              "description": "In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.",
              "director": "Martin Scorsese",
              "genre": "Crime",
              "id": 3,
              "isFavorite": false,
              "name": "Gangs of new york",
              "posterImage": "https://es31-server.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg",
              "previewImage": "https://es31-server.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 7.5,
              "released": 2002,
              "runTime": 167,
              "scoresCount": 370881,
              "starring": Array [
                "Leonardo DiCaprio",
                "Cameron Diaz",
                "Daniel Day-Lewis",
              ],
              "videoLink": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
            }
          }
          isPlaying={false}
          key="3"
        />
      </div>
    </section>
    <Footer />
  </div>
</PageWrapper>
`;
